proposal/supports-general.md

```diff
@@ -0,0 +1,167 @@
+# `@supports` `<general-enclosed>`: Draft 1.0
+
+*([Issue](https://github.com/sass/sass/issues/2780))*
+
+This proposal defines how Sass parses supports queries that use the
+[`<general-enclosed>`][] production.
+
+[`<general-enclosed>`]: https://drafts.csswg.org/mediaqueries-4/#typedef-general-enclosed
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+  * [Design Decisions](#design-decisions)
+    * [How Much To Support](#how-much-to-support)
+    * [SassScript Injection](#sassscript-injection)
+    * [Backwards Compatibility](#backwards-compatibility)
+* [Syntax](#syntax)
+  * [`SupportsCondition`](#supportscondition)
+  * [`InterpolatedAnyValue`](#interpolatedanyvalue)
+
+## Background
```

Comment:
It could be helpful to add more background on the current state of `<supports-decl>` in Sass, and why that complicates things.